salt/modules/status.py

```diff
@@ -148,26 +148,44 @@ def uptime():
 
         salt '*' status.uptime
     '''
-    ut_ret = {'seconds': 0}
+    curr_seconds = time.time()
+
+    # Get uptime in seconds
     if salt.utils.is_linux():
         ut_path = "/proc/uptime"
         if not os.path.exists(ut_path):
             raise CommandExecutionError("File {ut_path} was not found.".format(ut_path=ut_path))
-        ut_ret['seconds'] = int(float(open(ut_path).read().strip().split()[0]))
+        seconds = float(salt.utils.fopen(ut_path).read().split()[0])
     elif salt.utils.is_sunos():
         cmd = "kstat -p unix:0:system_misc:boot_time | nawk '{printf \"%d\\n\", srand()-$2}'"
-        ut_ret['seconds'] = int(__salt__['cmd.shell'](cmd, output_loglevel='trace').strip() or 0)
+        seconds = float(__salt__['cmd.shell'](cmd, output_loglevel='trace').strip() or 0)
```

Comment:
On Solaris, with `kstat` you will always get an integer, no need to `float` here.